lib/govuk_publishing_components/app_helpers/asset_helper.rb

```diff
@@ -79,12 +79,22 @@ def application_stylesheet_in_use?
     private
 
       def is_already_used?(component)
-        if GovukPublishingComponents::Config.exclude_css_from_static
+        if GovukPublishingComponents::Config.exclude_css_from_static && !viewing_component_guide?
           all_component_stylesheets_being_used.include?(component) || STATIC_STYLESHEET_LIST.include?(component)
         else
           all_component_stylesheets_being_used.include?(component)
         end
       end
+
+      def in_component_guide_path?
+        request.path.include?("/component-guide")
+      end
+
+      def viewing_component_guide?
+        return @viewing_component_guide if defined?(@viewing_component_guide)
+
+        @viewing_component_guide ||= in_component_guide_path?
+      end
```

Comment:
Is `@viewing_component_guide` likely to be defined outside this method? (ie is there a reason we're memoizing the value in this slightly non-idiomatic way)

Reply:
The main reason for using this approach was from following this blog post - https://blog.appsignal.com/2022/12/20/a-guide-to-memoization-in-ruby.html#ignoring-the-false-or-nil-return

In particular:

> Memoization using ||= doesn't consider false/nil return values, so such cases should be handled.

I also observed this when testing the changes in the `frontend` rendering app. In this case, `in_component_guide_path?` returns false as expected, but this method would be called several times, each returning a false value.

There could (is likely) a better way to do this, and one that makes sense to others reading the code.

Reply:
Actually, I think we might consider not memoizing at all? request.path.include? is just doing a string call, it probably doesn't need to be incredibly performant. It's not as necessary as memoizing a database or network call, or a complex calculation.

Reply:
Thanks Keith, good point. I've updated the code now and removed the memoization.